### PR TITLE
Implement CommandSpy hierarchy

### LINKED ISSUE #913: CommandSpy hierarchy
What about is super admins saw ops commands (As usual), though telnet admins and telnet.Clan admins see Super Admin commands, and seniors see Super Admin, Telnet and Telnet.Clan Admin commands, and mark (Or the owner) see's all commands.

### PR DESCRIPTION
**Pull request by LegendIsAwesomes**
Resolves #913

### CODE REVIEW

**Comment by LegendIsAwesomes** [comment]
@JeromSar

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 1]
```diff
@@ -733,6 +733,47 @@ public void onPlayerCommandPreprocess(PlayerCommandPreprocessEvent event)
...
+                }
+            }
+        
+        if (!TFM_AdminList.isSeniorAdmin(player))
+        {
+            for (Player pl : Bukkit.getOnlinePlayers())
+            {
+                if (TFM_AdminList.isSeniorAdmin(pl, true) && TFM_PlayerData.getPlayerData(pl).cmdspyEnabled())
+                {
+                    if (TFM_AdminList.isSuperAdmin(pl))
```
Comment (OxLemonxO) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: should change isSuperAdmin(pl)) to isTelnetAdmin(pl, true))

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 2]
```diff
@@ -733,6 +733,47 @@ public void onPlayerCommandPreprocess(PlayerCommandPreprocessEvent event)
                 }
             }
         }
+            if (!TFM_AdminList.isTelnetAdmin(player, true))
+            {
+                for (Player pl : Bukkit.getOnlinePlayers())
+                {
+                    if (TFM_AdminList.isTelnetAdmin(pl, true) && TFM_PlayerData.getPlayerData(pl).cmdspyEnabled())
+                    {
```
Comment (OxLemonxO) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: should be <code>
if (TFM_AdminList.isTelnetAdmin(pl, true) || TFM_AdminList.isSeniorAdmin(pl, true) && TFM_PlayerData.getPlayerData(pl).cmdspyEnabled()) "
</code>

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 3]
```diff
@@ -733,6 +733,47 @@ public void onPlayerCommandPreprocess(PlayerCommandPreprocessEvent event)
                 }
             }
         }
+            if (!TFM_AdminList.isTelnetAdmin(player, true))
```
Comment (OxLemonxO) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: should be if (!TFM_AdminList.isTelnetAdmin(player, true) || !TFM_AdminList.isSeniorAdmin(player, true) || !TFM_AdminList.SERVER_OWNERS.getList().contains(player))

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 4]
```diff
@@ -733,6 +733,47 @@ public void onPlayerCommandPreprocess(PlayerCommandPreprocessEvent event)
...
+                    }
+                }
+            }
+        
+        if (!TFM_AdminList.isSeniorAdmin(player))
+        {
+            for (Player pl : Bukkit.getOnlinePlayers())
+            {
+                if (TFM_AdminList.isSeniorAdmin(pl, true) && TFM_PlayerData.getPlayerData(pl).cmdspyEnabled())
+                {
```
Comment (OxLemonxO) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: should change to <code> if (TFM_AdminList.isSeniorAdmin(pl, true) || TFM_ConfigEntry.SERVER_OWNERS.getList().contains(pl) && TFM_PlayerData.getPlayerData(pl).cmdspyEnabled())

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 5]
```diff
@@ -733,6 +733,47 @@ public void onPlayerCommandPreprocess(PlayerCommandPreprocessEvent event)
...
+                        if (TFM_AdminList.isSuperAdmin(player))
+                            {
+                                TFM_Util.playerMsg(pl, player.getName() + ": " + command, ChatColor.YELLOW);
+                            }
+                    }
+                }
+            }
+        
+        if (!TFM_AdminList.isSeniorAdmin(player))
+        {
```
Comment (OxLemonxO) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: should change to <code> if (!TFM_AdminList.isSeniorAdmin(player) || !TFM_SERVER_OWNERS.getList().contains(pl))

File: src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java [hunk 6]
```diff
@@ -733,6 +733,47 @@ public void onPlayerCommandPreprocess(PlayerCommandPreprocessEvent event)
...
+        }
+        
+        if (!TFM_ConfigEntry.SERVER_OWNERS.getList().contains(player))
+        {
+            for (Player pl : Bukkit.getOnlinePlayers())
+            {
+                if (TFM_ConfigEntry.SERVER_OWNERS.getList().contains(pl) && TFM_PlayerData.getPlayerData(pl).cmdspyEnabled())
+                {
+                    if (TFM_AdminList.isSuperAdmin(player))
+                    {
```
Comment (OxLemonxO) [inline src/me/StevenLawson/TotalFreedomMod/Listener/TFM_PlayerListener.java]: should change to if (TFM_AdminList.isSeniorAdmin(player)

**Comment by LegendIsAwesomes** [comment]
Sorry used to using player not pl, will fix.

**Comment by OnlineRepz** [comment]
@LegendIsAwesomes, any updates?

**Comment by Wild1145** [comment]
There *must* be a cleaner way to manage the permissions rather than if's of if's of if's...

**Comment by JeromSar** [comment]
The associated issue has been closed.

**Pull request closed without merging**